syntax = "proto3";

import "google/protobuf/any.proto";
import "desmond/network/data_source.proto";

package desmond.perception;

message SensorDatum {
  // Unix timestamp when the datum was perceived.
  int64 time_usec = 1;

  // Payload of the sensor. Must match the type in the
  // SensorSpec
  google.protobuf.Any payload = 2;
}

message SensorSpec {
  network.DataSource data_source = 1;

  // Human-readable description of the sensory data.
  string description = 2;
}
